File: src/Level/Clue/adjacency_clue.py
from subclue import SubClue

class AdjacencyClue(SubClue):
    """ Represents the Adjacency Clue """
    
    def __init__(self):
        """ Initialize the Clue """
        self.numberOfAdjacentMines = 0
        self.reverse = False
    
    def update(self, minefield, gridRow, gridColumn):
        """ Update the subclue """
        self.numberOfAdjacentMines = 0
    
    def updateFromAdjacentSquare(self, square):
        """ Update the adjacency clue with an adjacent square """
        if square.mined():
            self.numberOfAdjacentMines += 1
            
    def getAdjacentMinesClue(self):
        """ Returns the Adjacent Mines Clue """
        if self.reverse:
            return 8 - self.numberOfAdjacentMines
        else:
            return self.numberOfAdjacentMines

File: src/android/src/com/kaottessur/pymine/level/clue/AdjacencyClue.java
package com.kaottessur.pymine.level.clue;

import com.kaottessur.pymine.level.GridSquare;
import com.kaottessur.pymine.level.Minefield;

public class AdjacencyClue extends SubClue {
	private int numberOfAdjacentMines = 0;
	
	public AdjacencyClue() {
		
	}
	
	@Override
	public void update(Minefield minefield, int gridRow, int gridColumn) {
		numberOfAdjacentMines = 0;
	}
	
	@Override
	public void updateFromAdjacentSquare(GridSquare gridSquare) {
		if (gridSquare.mined()) {
			numberOfAdjacentMines++;
		}
	}
	
	public int getClueReading() {
		if (reverse)
			return 8-numberOfAdjacentMines;
		else
			return numberOfAdjacentMines;
	}
}
